python/cudf/cudf/core/column/string.py

```diff
@@ -1786,6 +1791,66 @@ def translate(self, table, **kwargs):
             cpp_translate(self._column, table), **kwargs
         )
 
+    def normalize_spaces(self):
+        return cpp_normalize_spaces(self._column)
+
+    def tokenize(self, delimiter=" "):
+        delimiter = _massage_string_arg(delimiter, "delimiter", allow_col=True)
+        return cpp_tokenize(self._column, delimiter)
+
+    def token_count(self, delimiter=" "):
+        delimiter = _massage_string_arg(delimiter, "delimiter", allow_col=True)
+        return cpp_count_tokens(self._column, delimiter)
+
+    def ngrams(self, n=2, separator="_"):
+        separator = _massage_string_arg(separator, "separator")
+        return cpp_generate_ngrams(self._column, n, separator)
+
+    def ngrams_tokenize(self, n=2, delimiter=" ", separator="_"):
+        delimiter = _massage_string_arg(delimiter, "delimiter")
+        separator = _massage_string_arg(separator, "separator")
+
+        return cpp_ngrams_tokenize(self._column, n, delimiter, separator)
+
+
+def _massage_string_arg(value, name, allow_col=False):
+    from cudf._libxx.scalar import Scalar
+    from cudf._libxx.column import Column
+    from cudf.utils.dtypes import is_string_dtype
+
+    allowed_types = ["Scalar"]
+
+    if isinstance(value, str):
+        return Scalar(value, dtype="str")
+
+    if isinstance(value, Scalar) and is_string_dtype(value.dtype):
```

Comment:
If `value` is a non-string, non-Scalar type here, i.e. `bytes` or a `bytearray`. This will error with a no `dtype` attribute exception. Maybe it makes sense to first check if we have a `Scalar`, then check if we have a `numpy.str_`, otherwise call `Scalar(str(value), dtype="str")`?

Reply:
I'm confused where the no `dtype` attribute exception would be coming from. the `dtype` attribute is only accessed if the value is an instance of `Scalar`, no?

Reply:
My brain switched `and` --> `or`, this is good.